### PR TITLE
Add a failure analyzer for Hibernate bootstrap failure

### PR DESCRIPTION
**Pull request by govi20**
Added the failure analyzer for StrategySelectionException. Hibernate is throwing StrategySelectionException for the missing dialect value.

### CODE REVIEW

**Comment by philwebb** [comment]
@govi20 Thanks very much for looking into this. Ideally, before we merge this, we'd like to also get a test in place.

**Comment by govi20** [comment]
@philwebb Okay. I will be writing the test cases for the analyzer. Any idea why this build failed? I didn't make any change in FieldValues.java.

**Comment by philwebb** [comment]
I'm not sure why the CI failed, you could try rebasing on the latest `master`. The CI for that is currently green. Don't worry if it fails again, it looks unrelated to your work.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/diagnostics/analyzer/StrategySelectionExceptionAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * @author Govinda Sakhare
+ */
+public class StrategySelectionExceptionAnalyzer
+		extends AbstractFailureAnalyzer<StrategySelectionException> {
+
+	@Override
+	protected FailureAnalysis analyze(Throwable rootFailure,
+			StrategySelectionException strategySelectionException) {
+		return new FailureAnalysis(
+				"Dialect value is missing against database or incorrect value is provided!",
```
[the remark this answers is not in the document]
Comment (govi20) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/diagnostics/analyzer/StrategySelectionExceptionAnalyzer.java]: Thanks for flagging it out. I did not test the scenario's with old hibernate version.

[the remark this answers is not in the document]
Comment (snicoll) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/diagnostics/analyzer/StrategySelectionExceptionAnalyzer.java]: It's not so much about old Hibernate version but more about Hibernate failing to boostrap for whatever reason. A good way to move forward would be to contribute tests that are showcasing various cases where Hibernate could not start. We could then use that tests suite as a base for the failure analyzer. Would you be willing to update the PR in that direction?

[the remark this answers is not in the document]
Comment (govi20) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/diagnostics/analyzer/StrategySelectionExceptionAnalyzer.java]: Sure Stephane, I will start working in that direction.

**Comment by snicoll** [comment]
@govi20 thanks for the PR but this isn't going in the direction the original issue was opened against. I've removed the ideal for contribution label for this one as I believe we need to revisit our initial goal against the current state of affair. Thanks for your efforts, in any case!

**Pull request closed without merging**